composeApp/src/commonMain/kotlin/org/ooni/probe/ui/running/RunningViewModel.kt

```diff
@@ -10,23 +10,40 @@ import kotlinx.coroutines.flow.filterIsInstance
 import kotlinx.coroutines.flow.launchIn
 import kotlinx.coroutines.flow.onEach
 import kotlinx.coroutines.flow.update
+import org.ooni.probe.data.models.ProxySettings
+import org.ooni.probe.data.models.SettingsKey
 import org.ooni.probe.data.models.TestRunError
 import org.ooni.probe.data.models.TestRunState
-import org.ooni.probe.ui.dashboard.DashboardViewModel.Event
+import org.ooni.probe.data.repositories.PreferenceRepository
 
 class RunningViewModel(
     onBack: () -> Unit,
     goToResults: () -> Unit,
     observeTestRunState: Flow<TestRunState>,
     observeTestRunErrors: Flow<TestRunError>,
     cancelTestRun: () -> Unit,
+    preferencesRepository: PreferenceRepository,
 ) : ViewModel() {
     private val events = MutableSharedFlow<Event>(extraBufferCapacity = 1)
 
     private val _state = MutableStateFlow(State())
     val state = _state.asStateFlow()
 
     init {
+        preferencesRepository.allSettings(
+            listOf(
+                SettingsKey.PROXY_PROTOCOL,
+                SettingsKey.PROXY_HOSTNAME,
+                SettingsKey.PROXY_PORT,
+            ),
+        ).onEach { proxySettings ->
+            val proxy = ProxySettings.newProxySettings(
+                protocol = proxySettings[SettingsKey.PROXY_PROTOCOL] as? String,
+                hostname = proxySettings[SettingsKey.PROXY_HOSTNAME] as? String,
+                port = proxySettings[SettingsKey.PROXY_PORT] as? String,
+            ).getProxyString()
```

Comment:
This could be in a GetProxySettings domain class. That would make the code here simpler, and maybe it could be reused on `GetEnginePreferences` for example.